src/sentry/static/sentry/app/api.tsx

```diff
@@ -455,8 +460,15 @@ export class Client {
           getResponseHeader: (header: string) => response.headers.get(header),
         };
 
+        const responseContentType = response.headers.get('content-type');
+
+        // Respect the response content-type header
+        const responseData = responseContentType?.includes('json')
+          ? responseJSON
+          : responseText;
```

Comment:
When the status code is 204, this causes the response data to always be an empty string (i.e. `''`) which may be problematic in some cases where it's not expecting an empty string.

#24076 tries to address this for this one place but we may need to make sure this doesn't have an impact else where.